Microsoft.Azure.CosmosRepository/src/InMemoryRepository.cs

```diff
@@ -0,0 +1,147 @@
+        public async ValueTask<TItem> GetAsync(string id, PartitionKey partitionKey, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+
+            if (partitionKey == default)
+            {
+                partitionKey = new PartitionKey(id);
+            }
+
+            TItem item = Items.Values.FirstOrDefault(i => i.Id == id && new PartitionKey(i.PartitionKey) == partitionKey);
+
+            if (item is null)
+            {
+                NotFound();
+            }
+
+            return item is { Type: { Length: 0 } } || item.Type == typeof(TItem).Name ? item : default;
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<IEnumerable<TItem>> GetAsync(Expression<Func<TItem, bool>> predicate, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+            return Items.Values.Where(predicate.Compose(
+                item =>  item.Type == typeof(TItem).Name, Expression.AndAlso).Compile());
+        }
+
+        /// <inheritdoc/>
+        public ValueTask<IEnumerable<TItem>> GetByQueryAsync(string query, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc/>
+        public ValueTask<IEnumerable<TItem>> GetByQueryAsync(QueryDefinition queryDefinition, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<TItem> CreateAsync(TItem value, CancellationToken cancellationToken = default)
+        {
+            value.Id ??= Guid.NewGuid().ToString();
+
+            await Task.CompletedTask;
+
+            TItem item = Items.Values.FirstOrDefault(i => i.Id == value.Id && i.PartitionKey == value.PartitionKey);
+
+            if (item is not null)
+            {
+                Conflict();
+            }
+
+            Items.TryAdd(value.Id, value);
+
+            return value;
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<IEnumerable<TItem>> CreateAsync(IEnumerable<TItem> values, CancellationToken cancellationToken = default)
+        {
+            IEnumerable<TItem> enumerable = values.ToList();
+
+            foreach (TItem value in enumerable)
+            {
+                await CreateAsync(value, cancellationToken);
+            }
+
+            return enumerable;
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<TItem> UpdateAsync(TItem value, CancellationToken cancellationToken = default)
+        {
+            await Task.CompletedTask;
+
+            TItem item = Items.Values.FirstOrDefault(i => i.Id == value.Id && i.PartitionKey == value.PartitionKey);
+
+            if (item is not null)
+            {
+                Items.TryRemove(value.Id, out TItem _);
+            }
+
+            Items.TryAdd(value.Id, value);
+
+            return value;
```

Comment:
This could be simplified as:

```suggestion
            return Items[value.Id] = value;
```

Using the [built-in indexer](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.item?view=net-5.0), which will create or update accordingly.

Reply:
That's pretty neat was not aware of this.